pkg/fission-cli/cmd/spec/spec.go

```diff
@@ -276,9 +276,10 @@ func (fr *FissionResources) validateFunctionReference(functions map[string]bool,
 	return nil
 }
 
-func (fr *FissionResources) Validate(input cli.Input) error {
```

Comment:
I understand why returning the bool here. But I don't think it's a good way to do so.
Let's return a list of warning message `[]string` instead of `bool` to indicate there are warnings when applying spec.

Reply:
For example:
```
warnings := []string

warnings = append(warnings, <warn message>)

``

Reply:
We are already printing the warning on console. I thought there is no use of saving warnings again. I don't think we are going to use the warning again. @life1347

Reply:
oh, I mean instead of printing warning in `Validate()` we should return all of them. In this way, we are also able to write tests for the function by check returned msg instead of monitoring the console output.